noteable_magics/datasource_postprocessing.py

```diff
@@ -158,3 +163,58 @@ def postprocess_snowflake(
         db = dsn_dict['database']
 
         dsn_dict['database'] = f'{db}/{schema}'
+
+
+@register_postprocessor('sqlite')
+def postprocess_sqlite(
+    datasource_id: str, dsn_dict: Dict[str, str], create_engine_kwargs: Dict[str, Any]
+) -> None:
+    """Expect to have either a URL provided which we should download and place into TMPDIR as the database file,
+    or no such thing implying to run in memory mode.
+    """
+
+    if 'database' in dsn_dict:
+        cur_path = dsn_dict['database']
+        if cur_path == '' or cur_path == ':memory:':
+            # Empty path is alias for :memory:, and is fine.
+            return
+
+        # If it smells like a URL, we should download, stash it into a tmpfile,
+        # and respell dsn_dict['database'] to point to that file. Any exceptions in here
+        # will spoil the datasource.
+        parsed = urlparse(dsn_dict['database'])
+
+        # (Sigh, 'mock' as scheme due to cannot cleanly pytest requests-mock http or https urls
+        #  for reasons I trust from the requests-mocks docs)
+        if parsed.scheme in ('http', 'https', 'ftp', 'mock'):
+            max_download_time = 10  # seconds. What do we want this to be?
```

Comment:
Maybe settable via env var or magic config?

Reply:
K, gonna make it a controllable knob in the datasource configuration itself, so could come from the Gate-side datasource configuration.

Reply:
OK, done, and logged the about-to-attpemt-with max timeout. Dragged the gate-side log_capture fixture over here in the process.